### ISSUE TITLE
[16.0] product_variant_configurator: method _compute_price_rule crashes if products is an empty recordset

### ISSUE THREAD

**Issue by espo-tony** [OCA/product-variant#335]
Hi,
As by title, in module **product_variant_configurator** the `product.pricelist` method `_compute_price_rule` raises an exception IndexError: tuple index out of range when the parameter `products` is an empty recordset.

```
    def _compute_price_rule(self, products, qty, uom=None, date=False, **kwargs):
        """Overwrite for covering the case where templates are passed and a
        different uom is used."""
        if products[0]._name != "product.template":
            # Standard use case - Nothing to do
```

This scenario is considered to be possible also in the original Odoo method (check this: https://github.com/odoo/odoo/blob/4adda36260f3ba0fdf75688b419593006b80adb7/addons/product/models/product_pricelist.py#L129 ).
I suggest that a solution would be to check if products is empty, and if it is, just return the super (which in order will return an empty dictionary).

**Comment by pedrobaeza** [comment]
Steps to reproduce the problem without extra custom modules?

**Comment by espo-tony** [comment]
I cannot reproduce the issue with only Odoo CE modules. However a scenario where it might arise uses Odoo EE module **sale_temporal** , in which method `_compute_price_rule` returns this:

```
        return {
            **results,
            **super()._compute_price_rule(
                products - price_computed_products, qty, uom=uom, date=date, **kwargs),
        }
```
        
If `products` and `price_computed_products` contain the same records, then the super is called with an empty recordset.

**Comment by pedrobaeza** [comment]
The quality of EE modules is sometimes debatable... In any case, propose a PR doing such safeguard, and being harmless for the rest, it's not a problem. The only problem is that not being reproducible, that the case arises again in the future (or there are other side effects).

**Comment by Du-ma** [comment]
@espo-tony 
Fix over here >> #346.
@pedrobaeza 
I would argue that using `obj[0]._name` to extract a static value, In essence - is wrong.
The original method doesn't constrain itself with passed product object length at all, why constrain it here?

And we can't get any side-effects by fixing this, only way this could go wrong in any shape or form is if odoo did a big mistake of allowing mixing of different objecs.
